libs/component-library/src/lib/radio-button/RadioButton.tsx

```diff
@@ -1,105 +1,104 @@
+import {
+  createRestyleComponent,
+  createVariant,
+  VariantProps,
+} from '@shopify/restyle';
 import * as React from 'react';
-import { Animated, View, StyleSheet } from 'react-native';
+import { Animated, StyleSheet } from 'react-native';
+import { FxBox } from '../box/box';
+import { FxCheckIcon } from '../icons/icons';
 import {
   FxPressableOpacity,
   FxPressableOpacityProps,
 } from '../pressable-opacity/pressableOpacity';
+import { FxTheme } from '../theme/theme';
 import { useFxTheme } from '../theme/useFxTheme';
-
 import type { $Omit } from './../types';
 
-import { useRadioButtonContext } from './RadioButtonGroup';
+import { useRadioButtonContext, ValueType } from './RadioButtonGroup';
 import { handlePress, isChecked } from './utils';
 
-type RadioButtonProps = $Omit<FxPressableOpacityProps, 'children'> & {
-  /**
-   * Value of the radio button
-   */
-  value: string;
-  /**
-   * Status of radio button.
-   */
-  status?: 'checked' | 'unchecked';
-  /**
-   * Whether radio is disabled.
-   */
-  disabled?: boolean;
-  /**
-   * Function to execute on internal onPress.
-   */
-  onPress?: () => void;
-  /**
-   * Custom color for radio.
-   */
-  checkedColor?: string;
-  /**
-   * Custom color for unchecked radio.
-   */
-  uncheckedColor?: string;
-  /**
-   * Custom color for radio.
-   */
-  checkedDisabledColor?: string;
-  /**
-   * Custom color for unchecked disabled radio.
-   */
-  uncheckedDisabledColor?: string;
-  /**
-   * Custom color for unchecked disabled radio.
-   */
-  uncheckedDisabledBackgroundColor?: string;
-};
+const radioVariant = createVariant({
+  themeKey: 'radioVariants',
+  property: 'variant',
+});
+
+const FxRadioBase = createRestyleComponent<
+  React.ComponentProps<typeof Animated.View> &
+    VariantProps<FxTheme, 'radioVariants', 'variant'>,
+  FxTheme
+>([radioVariant], Animated.View);
+
+const radioCheckmarkVariant = createVariant({
+  themeKey: 'radioCheckmarkVariants',
+  property: 'type',
+});
 
+const FxRadioDot = createRestyleComponent<
+  React.ComponentProps<typeof FxBox> &
+    VariantProps<FxTheme, 'radioCheckmarkVariants', 'type'>,
+  FxTheme
+>([radioCheckmarkVariant], FxBox);
+
+type RadioButtonProps = React.ComponentProps<typeof FxRadioBase> &
+  $Omit<FxPressableOpacityProps, 'children'> & {
+    /**
+     * Value of the radio button
+     */
+    value: ValueType;
+    /**
+     * Status of radio button.
+     */
+    status?: 'checked' | 'unchecked';
+    /**
+     * Whether radio is disabled.
+     */
+    disabled?: boolean;
+    /**
+     * Function to execute on internal onPress.
+     */
+    onPress?: () => void;
+  };
+
+const RADIO_SIZE = 18;
+const CHECKBOX_SIZE = 20;
 const BORDER_WIDTH = 1;
 const BORDER_WIDTH_CHECKED = 6;
 
-/**
- * ## Usage
```

Comment:
Initially put this in as it may not have been obvious how to use the radio button, more so for the Group

Reply:
I guess I removed it because it didn't include the `RadioButtonGroup`. It's not really the correct use case without it. Do you want me to add it back in and swap out the `View` wrapper for a `RadioButtonGroup` wrapper?

Reply:
So there are two ways to use the RadioButton, either with RadioGroup controlling state for you or if you wish to pass your own state to it. I added in the Usage to show this. Feel free to add back in if you feel it adds value